src/main/java/org/mule/extension/db/internal/domain/connection/type/resolver/TypeResolver.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+
+package org.mule.extension.db.internal.domain.connection.type.resolver;
+
+import org.mule.extension.db.internal.domain.type.ResolvedDbType;
+
+import java.sql.SQLException;
+
+/**
+ * Resolve data type of the struct and arrays elements
+ *
+ * @since 1.5.2
+ */
+public interface TypeResolver {
```

Comment:
if this class is for resolving structs and arrays only, then it should be renamed